madara-prover-rpc-server/src/lib.rs

```diff
@@ -155,8 +212,17 @@ impl Prover for ProverService {
             trace,
         };
 
-        let prover_result =
+        let mut prover_result =
             call_prover(&execution_artifacts, &prover_config, &prover_parameters).await;
+
+        // If split proof was requested, build it
+        if build_split_proof == Some(true) && prover_result.is_ok() {
+            let res = verify_and_annotate_proof(prover_result.as_mut().expect("Result checked above")).await;
+            if res.is_err() {
+                prover_result = Err(res.err().unwrap());
+            }
+        };
```

Comment:
This lets us call verify/split after prove in an efficient way, and with minimal changes to the API.

It needs clean up, though.